packaging/examples/metrics/grafana-dashboards/strimzi-kafka.json

```diff
@@ -1115,10 +1115,10 @@
       "steppedLine": false,
       "targets": [
         {
-          "expr": "sum(jvm_memory_bytes_used{namespace=\"$kubernetes_namespace\",kubernetes_pod_name=~\"$strimzi_cluster_name-$kafka_broker\",strimzi_io_name=\"$strimzi_cluster_name-kafka\"}) by (kubernetes_pod_name)",
+          "expr": "sum(jvm_memory_bytes_used{namespace=\"$kubernetes_namespace\",kubernetes_pod_name=~\"$strimzi_cluster_name-$kafka_broker\",strimzi_io_name=\"$strimzi_cluster_name-kafka\"}) by (pod)",
           "format": "time_series",
           "intervalFactor": 1,
-          "legendFormat": "{{kubernetes_pod_name}}",
+          "legendFormat": "{{pod}}",
```

Comment:
why this change again, even considering you are filtering on `kubernetes_pod_name` attribute in the query?

Reply:
`kubernetes_pod_name` was not available in all queries for KafkaConnect AFAIR so that's the reason why I choose `pod`

Reply:
again the question here is ... why? More investigation needed from my perspective.

Reply:
Well, I expect that might be injected by our Prometheus configs. But the question is how it works on our example setup and not on OpenShift. Same as I said on #7243, I think we need to explain why is `pod` the right one and why does it not impact the other users. I think we need more then just _it works for me_.

When I tried the Kafka dashboard on my cluster, it seems to work fine. That said, ther eseem to be some inconsistences in the changes. For example this query:

```
sum(rate(jvm_gc_collection_seconds_sum{namespace=\"$kubernetes_namespace\",kubernetes_pod_name=~\"$strimzi_cluster_name-$kafka_broker\",strimzi_io_name=\"$strimzi_cluster_name-kafka\"}[5m])) by (pod)
```

=> This PR changes the grouping to be based on `pod` instead of `kubernetes_pod_name`. Yet, it seems to use `kubernetes_pod_name` in the `sum` command it self. So right now, I guess to make it work we actually need both `kubernetes_pod_name` and `pod` labels to be present. That seems weird. 

So I guess we need to figure out:
* What adds the `pod` and `kubernetes_pod_name` labels and where can they be expected
* Pick the best one and use it consistently

Reply:
It seems that we inject `kubernetes_pod_name` for each metric which is scraped from the strimzi related pods - https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/metrics/prometheus-install/strimzi-pod-monitor.yaml#L81 . So I guess `pod` is added by kubernetes/prometheus configuration. I will try to find some details.

Reply:
That actually makes sense, because `container_cpu_usage_seconds_total` looks to me like a kubernetes metric so that's the reason why it doesn't have `kubernetes_pod_name` in